### PR TITLE
DOC-98, DOC-114: PlatformInfo API changes JSDoc

### PR DESCRIPTION
**Pull request by ctrlaltdavid**
Updates JSDoc per recent changes in the PlatformInfo API.

Cases: 
- https://highfidelity.atlassian.net/browse/DOC-98
- https://highfidelity.atlassian.net/browse/DOC-114

Note: No executable code changes.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20587/HighFidelity-Beta-PR15988-0555cf1.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20587/HighFidelity-Beta-Interface-PR15988-0555cf1.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20587/HighFidelity-Beta-PR15988-0555cf1.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20587/HighFidelity-Quest-Beta-PR15988-0555cf1.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20587/HighFidelity-Beta-PR15988-0555cf1.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20587/HighFidelity-Beta-Interface-PR15988-0555cf1.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20746/HighFidelity-Beta-PR15988-5576bf7.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20746/HighFidelity-Beta-Interface-PR15988-5576bf7.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20746/HighFidelity-Beta-PR15988-5576bf7.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20746/HighFidelity-Beta-Interface-PR15988-5576bf7.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20746/HighFidelity-Beta-PR15988-5576bf7.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20746/HighFidelity-Quest-Beta-PR15988-5576bf7.apk)

File: libraries/platform/src/platform/backend/Platform.cpp [hunk 1]
```diff
@@ -10,19 +10,52 @@
...
 #include "../PlatformKeys.h"
 
+/**jsdoc
+ * Information on the computer platform as a whole.
+ * @typedef {object} PlatformInfo.PlatformDescription
+ * @property {PlatformInfo.ComputerDescription} computer - Information on the computer.
+ * @property {PlatformInfo.CPUDescription[]} cpus - Information on the computer's CPUs.
+ * @property {PlatformInfo.DisplayDescription[]} displays - Information on the computer's displays.
+ * @property {PlatformInfo.GPUDescription[]} gpus - Information on the computer's GPUs.
+ * @property {PlatformInfo.GraphicsAPIDescription[]} graphicsAPIs - Information on the computer's  graphics APIs.
```
Comment (jennaingersoll) [inline libraries/platform/src/platform/backend/Platform.cpp]:
```suggestion
 * @property {PlatformInfo.GraphicsAPIDescription[]} graphicsAPIs - Information on the computer's graphics APIs.
```

File: libraries/platform/src/platform/backend/Platform.cpp [hunk 2]
```diff
@@ -10,19 +10,52 @@
...
+ * @property {PlatformInfo.MemoryDescription} memory - Information on the computers memory.
```
Comment (jennaingersoll) [inline libraries/platform/src/platform/backend/Platform.cpp]:
```suggestion
 * @property {PlatformInfo.MemoryDescription} memory - Information on the computer's memory.
```

File: libraries/platform/src/platform/backend/Platform.cpp [hunk 3]
```diff
@@ -10,19 +10,52 @@
...
+ * @property {PlatformInfo.NICDescription} nics - Information on the computers network cards.
```
Comment (jennaingersoll) [inline libraries/platform/src/platform/backend/Platform.cpp]:
```suggestion
 * @property {PlatformInfo.NICDescription} nics - Information on the computer's network cards.
```

File: libraries/platform/src/platform/backend/Platform.cpp [hunk 4]
```diff
@@ -35,6 +68,45 @@ namespace platform { namespace keys {
...
+     * @property {string} [vendor] - If an OpenGL API, then the OpenGL vendor.
+     * @property {string} [shadingLanguageVersion] - If an OpenGL API, then the shading language version.
+     * @property {string[]} [extensions] - If an OpenGL API, then the list of OpenGL extensions supported.
+     *
+     * @property {PlatformInfo.VulkanAPIDescription[]} [devices] - If a Vulkan API, then the devices provided in the API.
+     */
+    /**jsdoc
+     * Information on a Vulkan graphics API.
+     * @typedef {object} PlatformInfo.VulkanAPIDescription
+     * @property {string}
```
Comment (jennaingersoll) [inline libraries/platform/src/platform/backend/Platform.cpp]: Remove?

File: libraries/platform/src/platform/backend/Platform.cpp (hunk 1 above)
Comment (ctrlaltdavid) [reply libraries/platform/src/platform/backend/Platform.cpp]: Done.

File: libraries/platform/src/platform/backend/Platform.cpp (hunk 2 above)
Comment (ctrlaltdavid) [reply libraries/platform/src/platform/backend/Platform.cpp]: Done.

File: libraries/platform/src/platform/backend/Platform.cpp (hunk 3 above)
Comment (ctrlaltdavid) [reply libraries/platform/src/platform/backend/Platform.cpp]: Done.

File: libraries/platform/src/platform/backend/Platform.cpp (hunk 4 above)
Comment (ctrlaltdavid) [reply libraries/platform/src/platform/backend/Platform.cpp]: No, this is used by the GraphicsAPIDescription type's "devices" property (i.e., the "vk" namespace in the code below this JSDoc.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20767/HighFidelity-Beta-PR15988-8ea4ee6.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20767/HighFidelity-Quest-Beta-PR15988-8ea4ee6.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20767/HighFidelity-Beta-PR15988-8ea4ee6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20767/HighFidelity-Beta-Interface-PR15988-8ea4ee6.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20767/HighFidelity-Beta-PR15988-8ea4ee6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20767/HighFidelity-Beta-Interface-PR15988-8ea4ee6.dmg

**Pull request merged**